### PR TITLE
Add IO operations in progress to be exported in diskio module

### PR DESCRIPTION
**Pull request by VertexC**
## What does this PR do?
Add `io.ops` in fields exported by system.diskio in metricbeat.
  
## Why is it important?
As elastic/beats#20691 points out, we may need to calculate the queue time based on this new field `system.diskio.io.ops`.

## Checklist
~- [ ] I have commented my code, particularly in hard-to-understand areas~
~- [ ] I have made corresponding changes to the documentation~
- [x] My code follows the style guidelines of this project
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in CHANGELOG.next.asciidoc or CHANGELOG-developer.next.asciidoc.

## Related issues
- Closes  elastic/beats#20691

### CODE REVIEW

**Comment by VertexC** [comment]
@andresrc Could you kindly review this PR?

**Comment by unixsurfer** [comment]
@VertexC I am sorry but I'm not familiar with the development of metricbeats and therefore I can't perform a  proper review.
I would like to thank you for taking the time to add support for IO operations in progress metric.

**Comment by VertexC** [comment]
> @VertexC I am sorry but I'm not familiar with the development of metricbeats and therefore I can't perform a proper review.
> I would like to thank you for taking the time to add support for IO operations in progress metric.

It is okay :) And thanks for posting this good-for-first issue.

File: metricbeat/module/system/diskio/diskio.go [hunk 1]
```diff
@@ -107,6 +107,7 @@ func (m *MetricSet) Fetch(r mb.ReporterV2) error {
 			},
 			"io": common.MapStr{
 				"time": counters.IoTime,
+				"ops":  counters.IopsInProgress,
```
Comment (fearful-symmetry) [inline metricbeat/module/system/diskio/diskio.go:110]: It looks like this metric is exclusive to linux. We're trying to move away from globally reporting metrics that are exclusive to one or two OSes, as it means that unsupported OSes will report an unhelpful `0`. We'll want to insert this into the MapStr later with a `runtime.GOOS == "linux"` guard.

File: metricbeat/module/system/test_system.py [hunk 2]
```diff
@@ -36,10 +36,10 @@
                                                "softirq.norm.pct", "steal.norm.pct", "system.norm.pct", "user.norm.pct"]
 
 SYSTEM_DISKIO_FIELDS = ["name", "read.count", "write.count", "read.bytes",
-                        "write.bytes", "read.time", "write.time", "io.time"]
+                        "write.bytes", "read.time", "write.time", "io.time", "io.ops"]
```
Comment (fearful-symmetry) [inline metricbeat/module/system/test_system.py:39]: Once we guard the `io.ops`, we'll want to remove this.

**Review by fearful-symmetry — CHANGES_REQUESTED** [review]

**Comment by VertexC** [comment]
The diskio_linux fails in CI due to timeout issue. I only have mac (unix) environment and relevant tests might just skip locally.  Do you have any idea on this timeout error? @fearful-symmetry 

```bash
=================================== FAILURES ===================================
____________________________ Test.test_diskio_linux ____________________________
self = <test_system.Test testMethod=test_diskio_linux>
    @unittest.skipUnless(re.match("(?i)linux", sys.platform), "os")
    def test_diskio_linux(self):
        """
        Test system/diskio output on linux.
        """
        self.render_config_template(modules=[{
            "name": "system",
            "metricsets": ["diskio"],
            "period": "5s"
        }])
        proc = self.start_beat()
>       self.wait_until(lambda: self.output_lines() > 0)
module/system/test_system.py:251: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <test_system.Test testMethod=test_diskio_linux>
cond = <function Test.test_diskio_linux.<locals>.<lambda> at 0x7f9365a1f510>
max_timeout = 10, poll_interval = 0.1, name = 'cond'
    def wait_until(self, cond, max_timeout=10, poll_interval=0.1, name="cond"):
        """
        Waits until the cond function returns true,
        or until the max_timeout is reached. Calls the cond
        function every poll_interval seconds.
    
        If the max_timeout is reached before cond() returns
        true, an exception is raised.
        """
        start = datetime.now()
        while not cond():
            if datetime.now() - start > timedelta(seconds=max_timeout):
                raise TimeoutError("Timeout waiting for '{}' to be true. ".format(name) +
>                                  "Waited {} seconds.".format(max_timeout))
E               beat.beat.TimeoutError: Timeout waiting for 'cond' to be true. Waited 10 seconds.
../libbeat/tests/system/beat/beat.py:363: TimeoutError
```

File: metricbeat/module/system/diskio/diskio.go [hunk 3]
```diff
@@ -110,6 +110,11 @@ func (m *MetricSet) Fetch(r mb.ReporterV2) error {
 			},
 		}
 
+		// Add linux-only ops in progress
+		if runtime.GOOS == "linux" {
+			event["io"].(common.MapStr)["ops"] = counters.IopsInProgress
```
Comment (fearful-symmetry) [inline metricbeat/module/system/diskio/diskio.go:115]: We probably shouldn't be casting like that. Try using `.Put()` instead.

**Review by fearful-symmetry — CHANGES_REQUESTED** [review]

**Comment by fearful-symmetry** [comment]
Not sure, I imagine it's a deeper failure that the CI is just reporting as a timeout.

**Comment by fearful-symmetry** [comment]
Not sure why we're running into CI issues here....

**Comment by fearful-symmetry** [comment]
jenkins, test this

**Comment by VertexC** [comment]
Hmmm, it reports there is one change requested but I don't see any unresolved requested changes. @fearful-symmetry

**Comment by VertexC** [comment]
@fearful-symmetry Hi, do you have any plan to merge this pr? I'm happy to merge and rebuil fields, but if it is not merged for certain time, new conflicts will occur.

**Comment by fearful-symmetry** [comment]
Hey, sorry about dropping this last week. We're good to go, it looks like. Just rebase the PR and I can approve it. @VertexC

**Comment by fearful-symmetry** [comment]
jenkins, test this

**Comment by VertexC** [comment]
It looks like Lint is somehow blocked. Shall I commit --amend and force push again? @fearful-symmetry

**Comment by fearful-symmetry** [comment]
Try to merge from master again, yah. Not sure why the CI is blocked now.

**Comment by fearful-symmetry** [comment]
jenkins, test this

**Comment by fearful-symmetry** [comment]
@VertexC while I fight with CI, one last thing: Can you add a changelog entry to `CHANGELOG.next.asciidoc` ?

**Comment by VertexC** [comment]
> @VertexC while I fight with CI, one last thing: Can you add a changelog entry to `CHANGELOG.next.asciidoc` ?

Done :)

**Comment by fearful-symmetry** [comment]
jenkins, test this

**Review by fearful-symmetry — APPROVED** [review]

**Comment by andresrc** [comment]
@fearful-symmetry does this need a backport?

**Pull request merged**